trace/config.go

```diff
@@ -16,25 +16,72 @@ package trace
 
 import "go.opencensus.io/trace/internal"
 
+// Default limits for the number of attributes, message events and links on each span
+// in order to prevent unbounded memory increase for long-running spans.
+// These defaults can be overriden with trace.ApplyConfig.
+// These defaults can also be overriden per-span by using trace.StartOptions
+// when creating a new span.
+// TODO: Add an annnoation limit when priorities are implemented.
+const (
+	DefaultMaxAttributes    = 32
+	DefaultMaxMessageEvents = 128
+	DefaultMaxLinks         = 32
+)
+
 // Config represents the global tracing configuration.
 type Config struct {
 	// DefaultSampler is the default sampler used when creating new spans.
 	DefaultSampler Sampler
 
 	// IDGenerator is for internal use only.
 	IDGenerator internal.IDGenerator
+
+	// The below config options must be set with a GlobalOption.
+	// maxAttributes sets a global limit on the number of attributes.
+	maxAttributes int
```

Comment:
I'm not sure these really belong in `trace.Config`. The simpler thing to do would be to just have them as package-vars. The reason to have them in `trace.Config` would be to be able to change them at arbitrary times at runtime in a goroutine-safe manner. But I can't really think of a compelling reason you'd want to do that. @rakyll interested to know what do you think of this?

Reply:
Discussed with @rakyll offline. We think the best approach is to remove the `GlobalOption` type and instead rely on `0` to mean "don't change" (just like it does for the other fields of `Config`. If the user wants to not buffer anything (alway drop) they can always set the field to -1 explicitly.